packages/react/src/runtimes/edge/EdgeChatAdapter.ts

```
@@ -85,22 +147,31 @@ export class EdgeChatAdapter implements ChatModelAdapter {
-      .pipeThrough(runResultStream());
+    try {
+      if (!result.ok) {
+        throw new Error(`Status ${result.status}: ${await result.text()}`);
+      }
 
-    let update: ChatModelRunResult | undefined;
-    for await (update of asAsyncIterable(stream)) {
-      yield update;
-    }
+      const stream = result
+        .body!.pipeThrough(streamPartDecoderStream())
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)